src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INDEX_OUT_OF_BOUNDS_MESSAGE
+            = "Index %d out of bounds for length %d";
+    private static final int ZERO_INDEX = 0;
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (first == null) {
+            first = last = new Node<>(null, value, null);
+        } else {
+            last.next = new Node<>(last, value, null);
+            last = last.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (first == null) {
+            indexValidation(index);
```

Comment:
we need to validate index only if first==null? What the difference between this check and `index == size`, maybe we could combine them?